llvm/include/llvm/IR/Cheri.h

```diff
@@ -55,6 +56,68 @@ inline bool isCheriPointer(Type *Ty, const DataLayout *DL) {
 }
 
 namespace cheri {
+
+/// Represents the LLVM-level attribute that is used to signal
+/// cross-compartment imports.
+class CapabilityImportAttr {
+public:
+  /// Parses the string value previously constructed into an instance of the
+  /// attribute object.
+  CapabilityImportAttr(StringRef ValueRef) {
```

Comment:
This could use some more comments. In particular, it'd be useful to document what the pre- and post- conditions are at the steps of parsing.

Reply:
https://github.com/CHERIoT-Platform/llvm-project/pull/147/commits/bdd4ef4a7eb2e0afe4bfaf13e952a0e1e9a79b9c explains a bit more what that function does and what it expects.